tests/product.py

```diff
@@ -96,5 +96,26 @@ def test_get_all_products(self):
         self.assertEqual(len(json_response), 3)
 
     # TODO: Delete product
+    def test_delete_product(self):
+        """
+        Ensure we can remove a product.
+        """
+        # add product
+        self.test_get_all_products()
+
+        # remove product
+        url = "/products/1"
+        data = {"id": 1}
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.delete(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
 
+        # get products and verify product was removed
+        url = "/products"
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+        response = self.client.get(url, None, format='json')
+        json_response = json.loads(response.content)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
This isn't testing that the product was deleted. Getting all the products should always return a 200 whether or not a product has been deleted. 

A better way to test it would be to try to get the product that was deleted or try to delete the product again

Reply:
fixed that and the error code.